lib/https.js

```diff
@@ -34,6 +34,12 @@ const { urlToOptions, searchParamsSymbol } = require('internal/url');
 function Server(opts, requestListener) {
   if (!(this instanceof Server)) return new Server(opts, requestListener);
 
+  if (typeof opts === 'function') {
+    requestListener = opts;
+    opts = undefined;
+  }
+  opts = opts ? util._extend({}, opts) : {};
```

Comment:
```js
> console.time('assign');for(var i=0; i < 1E7; ++i) { Object.assign({1:1, 2:2, 3:3}, {a:1, b:2, c:3}); };console.timeEnd('assign')
assign: 9825.217ms
undefined
> console.time('_extend');for(var i=0; i < 1E7; ++i) { u._extend({1:1, 2:2, 3:3}, {a:1, b:2, c:3}); };console.timeEnd('_extend')
_extend: 6174.819ms
undefined
```
so `util._extend` is ~ 60% of `Object.assign`.
IMHO if it's not better than 50%, `Object.assign` should be prefered